components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -197,7 +197,9 @@ public OAuth2ClientValidationResponseDTO validateClientInfo(String clientId, Str
                 validationResponseDTO.setPkceSupportPlain(appDO.isPkceSupportPlain());
                 return validationResponseDTO;
             } else {    // Provided callback URL does not match the registered callback url.
-                log.warn("Provided Callback URL does not match with the provided one.");
+                if (log.isDebugEnabled()) {
+                    log.debug("Provided Callback URL: " + callbackURI + " does not match with the provided one.");
```

Comment:
shouldn't the message say, provided callback does not match with registered ones?